### ISSUE TITLE
Multiple projects

### ISSUE THREAD

**Issue by cycomanic** [lantzproject/lantz#9]
Hi @hgrecco,

what has been the motivation behind splitting lantz into multiple repositories? I've just started to look into what changed in lantz with your reboot and have to say find the split somewhat confusing. If I want to look into the different aspects I will have to somehow combine them again into a single repository with submodules, which somewhat defeats the purpose. I thought the general advise for git repositories is to keep things that belong together (e.g. depend on each other) together (see e.g. this write-up https://tech.labs.oliverwyman.com/blog/2019/01/28/split-git/). I think you might make more difficult than necessary to get quick contributions.  Hope this does not come across overly critical I really like lantz.

**Comment by hgrecco** [comment]
> Hope this does not come across overly critical

Not at all. Well explained criticism is always helpful and allow for useful discussion. Let me first tell you that there is a clean and easy way to install everything in one go:

```
pip install lantzdev[full]
```
and you get the full thing (core + drivers + qt + ino + sims + dependencies except pyqt). I am also working in a conda installation
```
conda install -c hgrecco lantzfull
```
(still in testing) which not only install `lantzdev[full]` but also `cutelog`, `arduino-cli`, `pyqt`. This creates a really confortable environment to work. 

Now, going back to your question. I have found that splitting the packages enforced a cleaner separation between them and therefore maintainablity. That is the same reason I moved `pimpmyclass` outside. 

I will be very happy to make it more user friendly if the above solutions present any problem.

**Comment by cycomanic** [comment]
> Not at all. Well explained criticism is always helpful and allow for useful discussion. Let me first tell you that there is a clean and easy way to install everything in one go:

I was meaning more for development/quick drive-by patches. I find for someone not very familiar with the code-base, trying to get an understanding is harder when one first has to figure out which repositories to check out and look into. 
I (maybe this is really a personal preference) also find that using automated tools for e.g. refactoring is much easier if I have everything in the same directory. I don't need to go back and forth to look at different logs to figure out the related changes for changes across e.g. the core and drivers.

**Comment by hgrecco** [comment]
Sorry for the very long delay. I missed your response.

I use PyCharm for developement and I keep everything in one project. In that way, PyCharm is able to quickly jump between one package and the other without trouble. Additionally, I hope that if we have designed this correctly, not much refactoring would be necessary.

**Comment by bilderbuchi** [comment]
Fair point, using a full-blown IDE surely makes this easier (I love Pycharm for its refactoring capabilities!). 
However, let me chime in here and offer another anecdotal data point: Today I tried to look for/find something in the lantz core (the cached property implementation approach), and I was on the phone. In the browser, I found myself browsing/manually jumping from the lantz repo (luckily correctly guessing where to continue) to the lantz-core repo, and finally to pimpmyclass (luckily I knew to find it under your user profile, not lantzproject's), which was not that frictionless.

I readily concede, though, that collecting stuff separately in pimpmyclass really makes sense. Does having 6 separate repos for lantz really improve architectural separation (compared to just subpackages of lantz in one repo), if you end up tying them together in Pycharm to work, anyway?

Btw, I was wondering about your dev setup to work on the code: 
Above you advise to use `conda install -c hgrecco lantzfull` to install the packages to create a good work environment. However, does this not install the packaged packages somewhere in the anaconda installation, without git repos, and not in your home directory (or wherever your projects reside)? How would you then work on the code, and patch stuff that you could then propose upstream?
If I am correct, you'd have to manually clone all the git repos, `conda develop` or `pip install -e` all of them, and then put a pycharm project around them. Is this the way?

**Comment by bilderbuchi** [comment]
Btw, thanks for the conda feedstock/channel/packaging! I use Anaconda, and it's great to not have to drop down to pip to install stuff (with conda-forge, I nearly never have to reach for pip, nowadays).
